### PR TITLE
feat: roles [TA-2903]

### PR DESCRIPTION
**Pull request by thomas-gr4vy**
# Description

Adds the `/roles` and `/roles/assignments` APIs.

Re: [TA-2903](https://gr4vy.atlassian.net/browse/TA-2903)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

**Review by cbetta — APPROVED** [review]

File: reference/request-bodies/role_assignment_create_request.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: Role Assignment (Create)
+description: A request to create a role assignment.
+type: object
+x-tags:
+  - Roles
+
+properties:
```
Comment (szeying-lee) [inline reference/request-bodies/role_assignment_create_request.yaml:8]: Are any of these properties required / nullable?

Comment (thomas-gr4vy) [reply reference/request-bodies/role_assignment_create_request.yaml:8]: All properties are required (however, only `user` is currently supported for the `assignee_type`).

File: reference/openapi.yaml [hunk 2]
```diff
@@ -215,6 +215,11 @@ tags:
       In Gr4vy, a rule can be created that triggers actions anywhere in the payment flow.
     x-internal: true
 
+  - name: Roles
```
[the remark this answers is not in the document]
Comment (thomas-gr4vy) [reply reference/openapi.yaml:218]: @szeying-lee could you create a ticket for the anti-fraud services issue?

[the remark this answers is not in the document]
Comment (thomas-gr4vy) [reply reference/openapi.yaml:218]: @szeying-lee quite a few of the tags are out-of-order. I will address this separately.

File: reference/request-bodies/role_assignment_create_request.yaml (hunk 1 above)
Comment (szeying-lee) [reply reference/request-bodies/role_assignment_create_request.yaml:8]: We could specify that they're required, IIRC we can do this with this block:

```
required:
  - role
  - ...
```

File: reference/openapi.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (szeying-lee) [reply reference/openapi.yaml:218]: Created a ticket here: https://gr4vy.atlassian.net/browse/TA-2951

File: reference/paths/role_assignments__list-role-assignments.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+operationId: list-role-assignments
+summary: List role assignments
+description: |-
+  Returns a list of role assignments.
+  Role assignments can be filtered for a given role by providing the `role_id` search parameter,
+  or for a given assignee by providing both the `assignee_type` and `assignee_id` parameters.
```
Comment (cbetta) [inline reference/paths/role_assignments__list-role-assignments.yaml:7]:
```suggestion
  Returns a list of role assignments.

  Role assignments can be filtered for a given role by providing
  the `role_id` search parameter, or for a given assignee by providing
  both the `assignee_type` and `assignee_id` parameters.
```

**Review by cbetta — APPROVED** [review]

**Pull request merged**